grammar/grammar.y

```diff
@@ -212,6 +214,7 @@ PtrPayload <- PIPE ASTERISK? IDENTIFIER PIPE
 
 PtrIndexPayload <- PIPE ASTERISK? IDENTIFIER (COMMA IDENTIFIER)? PIPE
 
+PtrListPayload <- PIPE (ASTERISK? IDENTIFIER COMMA)* ASTERISK? IDENTIFIER PIPE
```

Comment:
Both the argument list and the payload allow trailing comma.

Reply:
the argument list does, but the payload doesn't currently compile with a trailing comma. This might be a bug in the compiler though

Reply:
Works for me?
```zig
comptime {
    const a = .{ 1, 2 };
    const b = .{ 1, 2 };
    const c = .{ 1, 2 };
    for (
        a,
        b,
        c,
    ) |
        ai,
        bi,
        ci,
    | {
        _ = ai;
        _ = bi;
        _ = ci;
        // foo
    }
}

```

Reply:
oh, that's a bit unfortunate. it works if you have more than one case, but if you only have one it doesn't compile:
```
comptime {
    const a = .{ 1, 2 };

    for (
        a,
    ) |
        ai,
    | {
        _ = ai;
        // foo
    }
}
```

Reply:
I'm thinking we should probably support the case where a single element for loop has a trailing comma in the spec, even if it errors in the compiler at the moment because it's probably a bug as it's different to most other places in the language. Does that sound reasonable?

Reply:
The trailing comma should be allowed regardless of the amount of items.

Reply:
It seems to be a bug in the `zig fmt` upgrade error. It can be fixed but it'll also be removed when 0.11.0 is tagged.

Reply:
Ok, it should now allow trailing commas for both, and disallow missing commas. Sorry for messing up so many times!